File: ObjectAvoidance/java/src/main/java/elite/math/trig/triangle/Adjacent.java
package elite.math.trig.triangle;

import elite.math.SquareRoot;

/**
 *
 * @author CFerg (Elite)
 */
public class Adjacent {

    /**
     *
     */
    public double adjacent;
    
    /**
     *
     */
    public Adjacent(){
        this.adjacent = 0.0;
    }
    
    /**
     *
     * @param h
     * @param o
     */
    public Adjacent(Hypotenuse h, Opposite o){
        this.adjacent = new SquareRoot((h.hypotenuse * h.hypotenuse) - (o.opposite * o.opposite)).squareRoot;
    }
}

File: ObjectAvoidance/src/elite/math/trig/triangle/Adjacent.kt
package elite.math.trig.triangle

import elite.math.SquareRoot

/**
 *
 * @author CFerg (Elite)
 */
class Adjacent {

    /**
     *
     */
    var adjacent: Double = 0.toDouble()

    /**
     *
     */
    constructor() {
        this.adjacent = 0.0
    }

    /**
     *
     * @param h
     * @param o
     */
    constructor(h: Hypotenuse, o: Opposite) {
        this.adjacent = SquareRoot(h.hypotenuse * h.hypotenuse - o.opposite * o.opposite).squareRoot
    }
}